### PR TITLE
Reject `Symmetric(Hermitian(A))` if diagonal of A is not real.

### PR DESCRIPTION
**Pull request by KlausC**
This solves issue JuliaLang/LinearAlgebra.jl#598

### CODE REVIEW

File: stdlib/LinearAlgebra/src/symmetric.jl [hunk 1]
```diff
@@ -162,9 +162,12 @@ for (S, H) in ((:Symmetric, :Hermitian), (:Hermitian, :Symmetric))
                 throw(ArgumentError("Cannot construct $($S); uplo doesn't match"))
             end
         end
-        $S(A::$H) = $S(A.data, sym_uplo(A.uplo))
+        $S(A::$H) = $S(A, sym_uplo(A.uplo))
         function $S(A::$H, uplo::Symbol)
             if A.uplo == char_uplo(uplo)
+                if $H == Hermitian && !isreal(diag(A.data))
```
Comment (fredrikekre) [inline stdlib/LinearAlgebra/src/symmetric.jl]:
```suggestion
                if $H === Hermitian && any(!isreal, A.data[i] for i in diagind(A.data))
```

File: stdlib/LinearAlgebra/src/symmetric.jl [hunk 2]
```diff
@@ -162,9 +162,12 @@ for (S, H) in ((:Symmetric, :Hermitian), (:Hermitian, :Symmetric))
...
+                    throw(ArgumentError("Cannot construct $($S)($($H))); requires real diagonal"))
```
Comment (fredrikekre) [inline stdlib/LinearAlgebra/src/symmetric.jl]:
```suggestion
                    throw(ArgumentError("Cannot construct $($S)($($H))); diagonal contains complex values"))
```

File: stdlib/LinearAlgebra/src/symmetric.jl (hunk 1 above)
Comment (KlausC) [reply stdlib/LinearAlgebra/src/symmetric.jl]: change has been included in next commit

File: stdlib/LinearAlgebra/src/symmetric.jl (hunk 2 above)
Comment (KlausC) [reply stdlib/LinearAlgebra/src/symmetric.jl]: change has been included in next commit

**Comment by KlausC** [comment]
I propose to merge this PR; there seem no objections.

**Review by fredrikekre — APPROVED** [review]

**Pull request merged**